submodules/radios/iridium.py

```diff
@@ -0,0 +1,252 @@
+import logging
+from threading import Lock
+from functools import partial
+
+from . import Radio
+from core import ThreadHandler
+
+from serial import Serial
+
+
+class Iridium(Radio):
+    def __init__(self, config):
+        """
+        Assumes Iridium is in low power mode on start. Sets up class fields.
+        :param config: the config dictionary loaded from config_default.yml
+        """
+        self.config = config
+        self.modules = dict()
+
+        self.logger = logging.getLogger("IRIDIUM")
+        self.read_lock = Lock()
+
+        self.serial = None
+        self.processes = {
+            "listen_thread": ThreadHandler(
+                target=partial(self.listen),
+                name="iridium-listen",
+                parent_logger=self.logger,
+            )
+        }
+
+    def start(self):
+        """
+           Opens the Iridium serial port and starts the listening thread.
+           Assumes enough power is present therefore the tty port exists.
+           If the Iridium check fails, it raises an error
+        """
+
+        self.serial = Serial(
+            self.config["iridium"]["serial_port"], baudrate=19200, timeout=30
+        )
+        self.serial.flush()
+
+        if self.check(5):
+            self.logger.debug("Iridium Check Successful")
+        else:
+            raise RuntimeError("Iridium Check Failed")
+
+        self.processes["listen_thread"].start()
```

Comment:
same issue as above. Iterate through `self.processes` instead of hardcoding one process